### PR TITLE
:tada:  Tool for generation catalog schema from OpenAPI definition file

### LINKED ISSUE #2652: Airbyte SDK: Generate HTTP stream catalog from OpenAPI spec
## Tell us about the problem you're trying to solve
One of the biggest pain points when writing HTTP connectors is creating JSON Schema from OpenAPI specs. We should provide a very easy way to go from OpenAPI spec to JsonSchema, either completely programmatically or via an external tool.

### PR DESCRIPTION
**Pull request by avida**
## What
Resolves #2652 

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: tools/openapi2jsonschema/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM python:3.7-slim
+
+RUN apt-get update && apt-get install -y git && rm -rf /var/lib/apt/lists/*
+
+RUN pip install git+https://github.com/avida/openapi2jsonschema.git
```
Comment (yevhenii-ldv) [inline tools/openapi2jsonschema/Dockerfile:5]: Perhaps it is worth specifying the tag or the version of the library that we install from GitHub in order to eliminate errors when updating the version of this library?

Comment (avida) [reply tools/openapi2jsonschema/Dockerfile:5]: This library has no releases, its my fork of openapi2jsonschema tool which is abandoned atm (last merged PR was about 1.5 years ago). Will create v0.1 tag

**Review by yevhenii-ldv — APPROVED** [review]

**Review by htrueman — APPROVED** [review]

**Review by Zirochkaa — APPROVED** [review]

File: tools/openapi2jsonschema/run.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+IMG_TAG="airbyte/openapi2jsonschema"
+
+test "$#" == "0" &&  {
+   echo "Please specify path to open api definition file"
+   exit
+}
+
+docker images | grep "$IMG_TAG"  > /dev/null || {
```
Comment (sherifnada) [inline tools/openapi2jsonschema/run.sh:10]: love the lazy instantiation here

**Review by sherifnada — APPROVED** [review]
Looks great! I have a couple of more requests to make this easily consumable by an Airbyte user. 

1. Can you add a note in the CDK docs advertising this tool maybe [here](https://docs.airbyte.io/connector-development/cdk-python/schemas). Something like a section which says that you can use this tool. 
2. Can you add some files to this directory `tools/openapi2jsonschema/examples` which are an example of generated files? e.g: 

```
examples/
  stripe-api/
    stripe-openapi-spec.json
    transactions.json
    ...
```

**Pull request merged**